### DISCUSSION TITLE
QSPI flash file system builds, but can't write to it!

### DISCUSSION THREAD

**Discussion by picarro-rdorfner** [micropython/micropython#15514] [Libraries & Drivers]
Hello folks.

I've tried searching for this in the forums but have not had any luck finding anything similar to the issues I'm having.

I have ported micro python to some custom hardware that uses the same IO devices as the Arduino Giga R1.  I started with the GIGA R1 port and have modified it to run on my hardware.  Which it does! However, I need more room than the internal flash of 128k so I'm not trying to get the QSPI flash working.  So far I've been able to get it to build, it shows up as a disk drive when I plug my custom hardware into my laptop, but I can't seem to format it, mount it or write to it. I've tried a few different file system formats, exFAT, FAT12 FAT16, but none seem to let me actually write data from my laptop onto it. I've disabled pretty much everything except for a few items in mpconfigboard.h..

#define MICROPY_FATFS_EXFAT         (1)
#define MICROPY_HW_ENABLE_RTC       (0)
#define MICROPY_HW_ENABLE_RNG       (0)
#define MICROPY_HW_ENABLE_ADC       (0)
#define MICROPY_HW_ENABLE_DAC       (0)
#define MICROPY_HW_ENABLE_USB       (1)
#define MICROPY_HW_HAS_SWITCH       (0)
#define MICROPY_HW_HAS_FLASH        (0)
#define MICROPY_HW_ENABLE_SERVO     (0)
#define MICROPY_HW_ENABLE_TIMER     (0)
#define MICROPY_HW_ENABLE_SDCARD    (1)
#define MICROPY_HW_ENABLE_MMCARD    (0)

// Flash storage config
#define MICROPY_HW_SPIFLASH_ENABLE_CACHE            (1)
#define MICROPY_HW_ENABLE_INTERNAL_FLASH_STORAGE    (0)

So at this point I'm really not sure what to try next. Any suggestions would be greatly appreciated.
Thanks
Rick

**Comment by AndreoBotelho** [comment]
Giga R1 has a 16mb external qspi flash mounted as filesystem so you need only copy the code and it will work as long as you hardware is working, first you need to check your hardware for the correct pin connections with QSPI hardware pins, a simple way is to disable qspi and use softSPI on micropython to test if the flash is working, mount, format and write some data to it.
